src/query.cpp

```diff
@@ -208,7 +208,10 @@ bool Query::edns(bool dnssec)
 
     // increment the counter with number of additional sections
     header->arcount = htons(ntohs(header->arcount)+1);
-    
+
+    // always set the "authenticated data" bit
+    header->ad = 1;
```

Comment:
shouldn't this be conditional on the dnssec?

Reply:
I removed it because the responder actually sets the bit. We only need to signal "dnssec OK" in the EDNS section. dnssec-aware resolvers will then set the AD bit if the data is authentic, and return RRSIGs if so.